sources/api/pluto/src/hyper_proxy/mod.rs

```diff
@@ -116,6 +123,20 @@ impl Proxy {
             force_connect: false,
         }
     }
+
+    /// Set `Proxy` authorization
+    pub fn set_authorization<C: Credentials + Clone>(&mut self, credentials: Authorization<C>) {
+        match self.intercept {
+            Intercept::Https => {
+                self.headers.typed_insert(ProxyAuthorization(credentials.0));
+            }
+            _ => {
+                self.headers
+                    .typed_insert(Authorization(credentials.0.clone()));
+                self.headers.typed_insert(ProxyAuthorization(credentials.0));
+            }
+        }
+    }
```

Comment:
This was lifted from [`hyper_proxy` source](https://docs.rs/hyper-proxy/latest/src/hyper_proxy/lib.rs.html#196-210) and it's not super clear to me either. In any case, we shouldn't be reaching this path since we're only proxying HTTPS traffic.